### ISSUE TITLE
Does optical flow need to be normalized first before sent into encoder?

### ISSUE THREAD

**Issue by Pixie412** [YueWuHKUST/FutureVideoSynthesis#3]
Hi,
thanks for providing code. Do you need to preprocess optical flow before send into encoder? e.g. normalizing optical flow value to (-1,1) or (0,1)?

**Comment by YueWuHKUST** [comment]
No, the optical flow is computed by PWCNet using ./back/models/pwcnet.py. And we call this optical flow generation in ./back/train.py Line 56-65. 


Before warping the last input frame using optical flow, the flow is normalized because torch.nn.functional.grid_sample requires the range to be [-1, 1]

**Comment by Pixie412** [comment]
Thanks for the reply. In the code, the optical flow is divided by (h-1)/2 and (w-1)/2, why not divided by (h-1) and (w-1)?

**Comment by YueWuHKUST** [comment]
The range of the grid is [-1,1]. And the optical flow represents the relative shift between corresponding pixels. So the range of optical flow should be 2. This preprocess follows the setting of Vid2Vid https://github.com/NVIDIA/vid2vid/blob/master/models/base_model.py#L193

**Comment by kathrinawu** [comment]
close because of inactivity
